generated/nifgen/session.py

```diff
@@ -1427,6 +1427,47 @@ def configure_standard_waveform(self, waveform, amplitude, frequency, dc_offset=
     def create_waveform(self, waveform_data_array):
         '''create_waveform
 
+        Creates an onboard waveform
+        for use in Arbitrary Waveform output mode or Arbitrary Sequence output
+        mode.
+
+        Note:
+        You must set OUTPUT_MODE to NIFGEN_VAL_OUTPUT_ARB or
+        NIFGEN_VAL_OUTPUT_SEQ before calling this function.
+
+        Tip:
+        This method requires repeated capabilities (usually channels). If called directly on the
+        nifgen.Session object, then the method will use all repeated capabilities in the session.
+        You can specify a subset of repeated capabilities using the Python index notation on an
+        nifgen.Session instance, and calling this method on the result.:
+
+            session['0,1'].create_waveform(waveform_data_array)
+
+        Args:
+            waveform_data_array (list of float): Array of data for the new arbitrary waveform. This may be an iterable of float, or for best performance a numpy.ndarray of dtype int16 or float64.
+
+        Returns:
+            waveform_handle (int): The handle that identifies the new waveform. This handle is used in other methods when referring to this waveform.
+        '''
+        try:
+            import numpy
+            numpy_imported = True
```

Comment:
Should we cache whether we can import numpy or not?

That way we don't take the hit of trying to import and throwing every time this function is called and numpy isn't installed.

Reply:
Good idea. We can do this:

* Move the import to Session.py (with try/except).
* When needed, check using `if 'numpy' in sys.modules:`

I'll make a note for its own PR.

Reply:
Wouldn't it be faster to set a global variable or even a class variable that is just a boolean? That way we don't even have to do a string lookup in a list each time.